packages/dds/tree/src/test/feature-libraries/editableTree.spec.ts

```diff
@@ -0,0 +1,496 @@
+    const forest = buildForest(schemaRepo);
+    initializeForest(forest, data);
+    return forest;
+}
+
+function buildTestProxy(data: JsonableTree): UnwrappedEditableField {
+    const forest = setupForest(fullSchemaData, [data]);
+    const [context, field] = getEditableTree(forest);
+    return field;
+}
+
+function buildTestPerson(): PersonType {
+    const proxy = buildTestProxy(person);
+    return proxy as PersonType;
+}
+
+function expectTreeEquals(inputField: UnwrappedEditableField, expected: JsonableTree): void {
+    assert(inputField !== undefined);
+    assert(!Array.isArray(inputField));
+    // Above assert fails to narrow type to exclude readonly arrays, so cast manually here:
+    const node = inputField as EditableTreeOrPrimitive;
+    if (isPrimitiveValue(node)) {
+        // UnwrappedEditableTree loses type information (and any children),
+        // so this is really all we can compare:
+        assert.equal(node, expected.value);
+        return;
+    }
+    // Confirm we have an EditableTree object.
+    assert(node[proxySymbol] !== undefined);
+    assert.equal(node[valueSymbol], expected.value);
+    const type = node[typeSymbol];
+    const expectedType = schemaMap.get(expected.type) ?? fail("missing type");
+    assert.equal(type, expectedType);
+    for (const key of Object.keys(node)) {
+        const subNode: UnwrappedEditableField = node[key];
+        assert(subNode !== undefined, key);
+        const fields = expected.fields ?? {};
+        assert.equal(key in fields, true);
+        const field: JsonableTree[] = fields[key];
+        const isSequence = getFieldKind(getFieldSchema(type, key)).multiplicity === Multiplicity.Sequence;
+        // implicit sequence
+        if (isSequence) {
+            expectTreeSequence(subNode, field);
+        // explicit array
+        } else if (Array.isArray(subNode)) {
+            // That's actually not needed, as normally tree "users" would just recognize an array,
+            // but it might be a good memo for the primary field topic + see "TODO" in `getPrimaryField`
+            const primary = getPrimaryField(schemaMap.get(field[0].type) ?? fail("missing type"));
+            if (primary !== undefined) {
+                // Handle inlined primary fields
+                const expectedField = field[0].fields?.[primary.key];
+                assert(expectedField !== undefined);
+                expectTreeSequence(subNode, expectedField);
+            }
+        } else {
+            assert.equal(field.length, 1);
+            expectTreeEquals(subNode, field[0]);
+        }
+    }
+}
+
+function expectTreeSequence(field: UnwrappedEditableField, expected: JsonableTree[]): void {
+    assert(Array.isArray(field));
+    assert(Array.isArray(expected));
+    assert.equal(field.length, field.length);
+    for (let index = 0; index < field.length; index++) {
+        expectTreeEquals(field[index], expected[index]);
+    }
+}
+
+describe.only("editable-tree", () => {
```

Comment:
Gott make sure we don't check this in. Would skip all other tests in the package! (Looks like I left that in in my changes! Not good.)

```suggestion
describe("editable-tree", () => {
```